### DISCUSSION TITLE
How to restore closed indices ? (No alerts showing from past traffic capture)

### DISCUSSION THREAD

**Discussion by samqadah** [Security-Onion-Solutions/securityonion#2893] [Unsupported Versions]
Greetings,

Past alerts are no longer showing, I am assuming they no longer show since curator's default config is to close indices older than 30 days by default. I would like to restore (open) these closed indices. Following the documentation here:
https://www.elastic.co/guide/en/elasticsearch/client/curator/current/ex_open.html
I created the open.yml config file and placed it in the /opt/so/conf/curator/actions directory where other actions are also stored. I restarted curator but still no alerts are showing in the alerts tab. I can see the captured traffic on kibana though and everything is okay in terms of cpu utilization and memory usage on grifana.

I have a standalone deployment which im using for a one-time traffic analysis, I have already gathered traffic using SPAN port mirroring. 

Could anyone please guide me through on how to restore the closed indices. Much appreciated.

**Comment by samqadah** [comment]
Please note that I have edited the curator close age to 100 days in the directory:
/opt/so/saltstack/local/pillar/minions/$SENSORNAME_$ROLE.sls

**Reply by cm-ops** [reply]
Try placing the open.yml in `/opt/so/saltstack/local/salt/curator/files/action/`. Salt will copy it to `/opt/so/conf/curator/action`. Once you have the action file loaded you can check the curator log in `/opt/so/log/curator/` to see if the indices are being opened.

**Comment by cyb3rslack3r** [comment]
If your still trying to solve this, I had luck in SO16 using: curl -XPOST 'http://localhost:9200/*index_name*/_open'

If you are looking for closed indexes, you can find them by using: curl 'http://localhost:9200/_cluster/state/blocks?pretty=true'

I think in the past you could restore all in one swoop, but I could only manage to restore one index file at a time.  It was a pain, but better then not having the data

Hopefully this helps

**Reply by weslambert** [reply]
You could do something like the following to close all of them:

`for i in $(curl -s localhost:9200/_cat/indices | grep close | awk '{print $3}'); do curl -XPOST localhost:9200/$i/_open ; done`

Keep in mind, that Curator will come back and close them if you have not already adjusted close days.
